control-plane-operator/controllers/hostedcontrolplane/ocm/deployment.go

```diff
@@ -75,6 +75,10 @@ func ReconcileDeployment(deployment *appsv1.Deployment, ownerRef config.OwnerRef
 		util.BuildVolume(ocmVolumeServingCert(), buildOCMVolumeServingCert),
 		util.BuildVolume(ocmVolumeKubeconfig(), buildOCMVolumeKubeconfig),
 	}
+	// The OCM takes up to 75 seconds to finish its graceful shutdown, give it enough time
+	// to do that + 15 seconds margin -> 90s. The shutdown sequence is described in detail here:
+	// https://github.com/openshift/installer/blob/master/docs/dev/kube-apiserver-health-check.md
+	deployment.Spec.Template.Spec.TerminationGracePeriodSeconds = pointer.Int64Ptr(90)
```

Comment:
How is that doc related to the problem this PR solves? All we try to achieve here is to prevent abrupt termination of pods that take longer than expected to shut down.

Reply:
> How is that doc related to the problem this PR solves? 

You tell me, you added it? I asked to link to a source for why we need this, not to link to the unrelated kube-apiserver graceful shutdown documentation

Reply:
Removed the unrelated link from comments. As explained in the comments, in real hypershift deployments it takes more than 30 seconds to gracefully shutdown instances of those pods, that's the reason why this change is needed, confirmed by tests.

Reply:
You copied a bunch of unrelated docs over with an explanation that is plain wrong in this context.

In my self-hosted OCP cluster I can see that the OAS has a 70 second terminationGracePeriodSecond and the OCM the default 30 seconds. Please reference the code from the responsible cluster operators in your changes. If you derive from what they do, please explain what problem that solves and provide a reproducible testcase for that problem.

Reply:
Hey, just to provide a bit more info: IBM ROKS sets `terminationGracePeriodSeconds` to 90 seconds for these 2 components to minimize dropped connections when performing operations like roll outs and master refreshes. Especially on high-load, we frequently reached the grace period in the past, which resulted in abrupt disconnects.

Could we potentially make this value configurable, or patch-able? We would like to continue using our tuned values, if possible.